skafka/src/main/scala/com/evolutiongaming/skafka/ClientMetricsProvider.scala

```diff
@@ -0,0 +1,37 @@
+package com.evolutiongaming.skafka
+
+import cats.effect.kernel.Sync
+import org.apache.kafka.clients.consumer.Consumer
+import org.apache.kafka.clients.producer.Producer
+import org.apache.kafka.common.{Metric, MetricName}
+
+import java.util
+import scala.jdk.CollectionConverters._
+
+private[skafka] trait ClientMetricsProvider[F[_]] {
+  def metricsValues: Seq[ClientMetric[F]]
+}
+private[skafka] object ClientMetricsProvider {
+
+  def apply[F[_]: Sync](consumer: Consumer[_, _]): ClientMetricsProvider[F] =
+    new ClientMetricsProviderImpl(consumer.metrics())
+
+  def apply[F[_]: Sync](producer: Producer[_, _]): ClientMetricsProvider[F] =
+    new ClientMetricsProviderImpl[F](producer.metrics())
+
+  private class ClientMetricsProviderImpl[F[_]: Sync](source: => util.Map[MetricName, _ <: Metric])
```

Comment:
why do we need this ` _ <: Metric` and not just `Metric` ?

Reply:
can we just use `Map<MetricName, Metric> metrics();` ?

Reply:
no, it won't compile